units/provisioning_create_host.go

```diff
@@ -216,9 +216,22 @@ func (j *createHostJob) Run(ctx context.Context) {
 	}
 
 	defer func() {
-		if j.RetryInfo().GetRemainingAttempts() == 0 && j.HasErrors() && (j.host.Status == evergreen.HostUninitialized || j.host.Status == evergreen.HostBuilding) && j.host.SpawnOptions.SpawnedByTask {
-			if err := task.AddHostCreateDetails(j.host.StartedBy, j.host.Id, j.host.SpawnOptions.TaskExecutionNumber, j.Error()); err != nil {
-				j.AddError(errors.Wrapf(err, "adding host create error details"))
+		if j.RetryInfo().GetRemainingAttempts() == 0 && j.HasErrors() && (j.host.Status == evergreen.HostUninitialized || j.host.Status == evergreen.HostBuilding) {
+			if j.host.SpawnOptions.SpawnedByTask {
+				if err := task.AddHostCreateDetails(j.host.StartedBy, j.host.Id, j.host.SpawnOptions.TaskExecutionNumber, j.Error()); err != nil {
+					j.AddError(errors.Wrapf(err, "adding host create error details"))
+				}
+			}
+			if j.host.UserHost {
+				// Terminate spawn hosts immediately if they have no creation
```

Comment:
I'm a little bit worried that if some is not subscribed to receive notifications for spawn host outcome, there will be no feedback indicating that something went wrong and it will simply disappear. 
However, I don't have any better solutions (because sending a notification without them being subscribed is not a good idea), and it's still better than the current setup.

Reply:
If they're subscribed, they'll receive a notification regardless of whether the host gets terminated here or later (when it hits the stale timeout), so they do get feedback as long as they're subscribed. The issue is that the host may be terminated before they get the notification and before Spruce displays the event log, which may result in a couple minutes of confusion on where their requested spawn host went. Regardless, they can also see their terminated host in the UI for a few minutes after it's terminated.

I can remove this for now since the spawn host will eventually go away, it'll just take 15 minutes of waiting.

Reply:
I'm less worried about the couple of minutes of confusion, I'm more worried about the person who is not subscribed to notifications. Because those people before had the extra step of terminating it themselves so they knew what was happening, but now I thought there would be no feedback or sign of what's happening. 

But I had forgotten that they can still see their terminated host in the UI for a few minutes after it's terminated. In that case I'm less worried about it.